syntax = "proto3";

package counter;

message CreateRequest {
    string counter_name = 1;
}

message CreateResponse {}

message IncrementRequest {
    string counter_name = 1;
    int64 amount = 2;
}

message IncrementResponse {}

message GetRequest {
    string counter_name = 1;
}

message GetResponse {
    int64 current_value = 1;
}

message GetCurrentValuesRequest {}

message CounterValue {
    string counter_name = 1;
    int64 current_value = 2;
}

message GetCurrentValuesResponse {
    repeated CounterValue values = 1;
}

service CounterService {
    rpc Create(CreateRequest) returns (CreateResponse) {}

    rpc Increment(IncrementRequest) returns (IncrementResponse) {}

    rpc Get(GetRequest) returns (GetResponse) {}

    rpc GetCurrentValues(GetCurrentValuesRequest) returns (GetCurrentValuesResponse) {}
}
